pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -685,6 +689,10 @@ func setSafeDefaults(spec *PerconaXtraDBClusterSpec, log logr.Logger) {
 		loginfo("Cluster size will be changed from %d to %d due to safe config", spec.PXC.Size, spec.PXC.Size+1)
 		spec.PXC.Size++
 	}
+
+	if spec.ProxySQL.Size < minSafeProxySQLSize {
```

Comment:
We need to add check if `spec.ProxySQL` is not `nil` and enabled =)

Reply:
Fixed in e6ab4ee9c0639938998154c8275c7e6564df8a08

Reply:
We also need to check if proxysql is enabled and log message according docs, if we set safe size